File: Emulator/src/main/content/global/skill/slayer/npc/CaveBugNPC.java
package content.global.skill.slayer.npc;

import core.game.node.entity.Entity;
import core.game.node.entity.npc.AbstractNPC;
import core.game.node.entity.player.Player;
import core.game.world.map.Location;
import core.plugin.Initializable;
import org.rs.consts.NPCs;

/**
 * The type Cave bug npc.
 */
@Initializable
public class CaveBugNPC extends AbstractNPC {

    /**
     * Instantiates a new Cave bug npc.
     */
    public CaveBugNPC() {
        super(0, null);
    }

    /**
     * Instantiates a new Cave bug npc.
     *
     * @param id       the id
     * @param location the location
     */
    public CaveBugNPC(int id, Location location) {
        super(id, location);
    }

    @Override
    public void finalizeDeath(Entity killer) {
        super.finalizeDeath(killer);
        if (killer instanceof Player) {
            Player p = (Player) killer;
        }
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new CaveBugNPC(id, location);
    }

    @Override
    public int[] getIds() {
        return new int[]{
            NPCs.CAVE_BUG_1832,
            NPCs.CAVE_BUG_LARVA_1833,
            NPCs.CAVE_BUG_5750
        };
    }
}

File: game-server/src/main/content/global/skill/slayer/npc/CaveBugNPC.kt
package content.global.skill.slayer.npc

import content.global.skill.slayer.Tasks
import core.game.node.entity.Entity
import core.game.node.entity.npc.AbstractNPC
import core.game.node.entity.player.Player
import core.game.world.map.Location
import core.plugin.Initializable

/**
 * The type Cave bug npc.
 */
@Initializable
class CaveBugNPC : AbstractNPC {
    /**
     * Instantiates a new Cave bug npc.
     */
    constructor() : super(0, null)

    /**
     * Instantiates a new Cave bug npc.
     *
     * @param id       the id
     * @param location the location
     */
    constructor(id: Int, location: Location?) : super(id, location)

    override fun finalizeDeath(killer: Entity) {
        super.finalizeDeath(killer)
        if (killer is Player) {
            val p = killer
        }
    }

    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return CaveBugNPC(id, location)
    }

    override fun getIds(): IntArray {
        return Tasks.CAVE_BUG.npcs
    }
}
